File: client/iih_stub/iih.ci_stub/iih.ci.ord/s/CiOrdVerifyServiceImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iih.ci.ord.dto.orderverify.d;
using xap.mw.coreitf.d;
using xap.mw.serviceframework;
using xap.rui.appfw;

namespace iih.ci.ord.i
{
    public class ICiOrdVerifyServiceImpl : ICiOrdVerifyService
    {
        private string url_r = XapSvrConfig.BaseUrl + "iihci.ord/iih.ci.ord.i.ICiOrdVerifyService";//ConfigUtil.getServiceUrl();
        
        private ServiceInvocation si;
		
        /// <summary>
        /// 构造函数
        /// </summary>
        public ICiOrdVerifyServiceImpl()
        {
            si = new ServiceInvocationImpl();
            si.url = url_r;
        }

        /// <summary>
        /// 医嘱审核检索患者列表
        /// </summary>
        /// <param name="cond"></param>
        /// <returns></returns>
        public OrderVerifyPatDTO[] getVerifyPat(OrderVerifyCondDTO cond)
        {
            List<object> param = new List<object>();
            param.Add(cond);
            si.url = url_r;
            var result = si.invokeList<OrderVerifyPatDTO>("getVerifyPat", param.ToArray());
            return result;
        }

        /// <summary>
        /// 通过医嘱id检索医嘱信息
        /// </summary>
        /// <param name="idOrs"></param>
        /// <returns></returns>
        public OrderVerifyDTO[] getPatOrderByIds(String idOrs)
        {
            List<object> param = new List<object>();
            param.Add(idOrs);
            si.url = url_r;
            var result = si.invokeList<OrderVerifyDTO>("getPatOrderByIds", param.ToArray());
            return result;
        }

        /// <summary>
        /// 系统自动审核医嘱
        /// </summary>
        /// <param name="orderIds"></param>
        /// <returns></returns>
        public OrderVerifyDTO[] verifyOrderBySys(string[] orderIds) 
        {
            List<object> param = new List<object>();
            param.Add(orderIds);
            si.url = url_r;
            var result = si.invokeList<OrderVerifyDTO>("verifyOrderBySys", param.ToArray());
            return result;
        }

        /// <summary>
        /// 医嘱审核确认
        /// </summary>
        /// <param name="orders"></param>
        public void verifyOrder(OrderVerifyDTO[] orders) 
        {
            List<object> param = new List<object>();
            param.Add(orders);
            si.url = url_r;
            si.invoke<string>("verifyOrder", param.ToArray());
        }

        /// <summary>
        /// 医生核对异常医嘱
        /// </summary>
        /// <param name="ordId"></param>
        /// <param name="ifAccept"></param>
        /// <param name="explain"></param>
        public void confirmVerifyResult(string ordId, FBoolean ifAccept, string explain) 
        {
            List<object> param = new List<object>();
            param.Add(ordId);
            param.Add(ifAccept);
            param.Add(explain);
            si.url = url_r;
            si.invoke<string>("confirmVerifyResult", param.ToArray());
        }
    }
}

File: server/iih.ci/iih.ci.ord/src/main/java/iih/ci/ord/s/CiOrdVerifyServiceImpl.java
package iih.ci.ord.s;

import iih.ci.ord.dto.orderverify.d.OrderVerifyCondDTO;
import iih.ci.ord.dto.orderverify.d.OrderVerifyDTO;
import iih.ci.ord.dto.orderverify.d.OrderVerifyPatDTO;
import iih.ci.ord.i.ICiOrdVerifyService;
import iih.ci.ord.s.bp.ordverify.ConfirmVerifyResultBP;
import iih.ci.ord.s.bp.ordverify.GetOpPatOrderBP;
import iih.ci.ord.s.bp.ordverify.GetOrderByIdsBP;
import iih.ci.ord.s.bp.ordverify.VerifyOrderBP;
import iih.ci.ord.s.bp.ordverify.VerifyOrderBySysBP;
import xap.mw.core.annotation.Service;
import xap.mw.core.data.BizException;
import xap.mw.core.service.constant.Binding;
import xap.mw.coreitf.d.FBoolean;

/**
 * 医嘱审核服务
 * @author ly
 *
 */
@Service(serviceInterfaces={ICiOrdVerifyService.class}, binding=Binding.JSONRPC)
public class CiOrdVerifyServiceImpl implements ICiOrdVerifyService {

	/**
	 * 医嘱审核检索患者列表
	 * @param cond 检索条件
	 * @return
	 * @throws BizException
	 */
	@Override
	public OrderVerifyPatDTO[] getVerifyPat(OrderVerifyCondDTO cond) throws BizException{
		GetOpPatOrderBP bp = new GetOpPatOrderBP();
		return bp.exec(cond);
	}
	
	/**
	 * 通过医嘱id检索医嘱信息
	 * @param idOrs
	 * @return
	 * @throws BizException
	 */
	@Override
	public OrderVerifyDTO[] getPatOrderByIds(String idOrs) throws BizException{
		GetOrderByIdsBP bp = new GetOrderByIdsBP();
		return bp.exec(idOrs);
	}
	
	/**
	 * 系统自动审核医嘱
	 * @param orderIds
	 * @return 系统审核信息
	 * @throws BizException
	 */
	@Override
	public OrderVerifyDTO[] verifyOrderBySys(String[] orderIds) throws BizException {
		VerifyOrderBySysBP bp = new VerifyOrderBySysBP();
		return bp.exec(orderIds);
	}

	/**
	 * 医嘱审核确认
	 * @param orders
	 * @throws BizException
	 */
	@Override
	public void verifyOrder(OrderVerifyDTO[] orders) throws BizException {
		VerifyOrderBP bp = new VerifyOrderBP();
		bp.exec(orders);
	}

	/**
	 * 医生核对异常医嘱
	 * @param ordId
	 * @param ifAccept
	 * @param explain
	 * @throws BizException
	 */
	@Override
	public void confirmVerifyResult(String ordId,FBoolean ifAccept,String explain) throws BizException{
		ConfirmVerifyResultBP bp = new ConfirmVerifyResultBP();
		bp.exec(ordId, ifAccept, explain);
	}
}
